pkg/cluster/k8sres.go

```diff
@@ -763,6 +764,14 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*v1beta1.State
 		return nil, fmt.Errorf("could not generate resource requirements: %v", err)
 	}
 
+	if spec.InitContainers != nil {
+		if c.OpConfig.EnableInitContainers {
+			initContainers = spec.InitContainers
+		} else {
+			c.logger.Warningf("InitContainers specified but globally disabled!")
```

Comment:
IMHO it is an example of invalid manifest and it should be rejected, instead of accepted to processing but with fine print in logs that no init containers are to be created. when users create PG object and see `status: Running` in it they assume it is all good and ready, but it is not what they asked